### PR TITLE
Fixes #886 Create parameter

### LINKED ISSUE #886: Create parameters using the diagram editor
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-IME
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-IME-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-IME-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
| Done | Requirement | Description |
| --- | --- | --- |
| <ul><li>[ ] </li></ul> | GCD-FSR-0720 | The Architecture Definition Diagram editor must allow dropping ParameterType objects onto ElementDefinition objects to create Parameters within those ElementDefinitions.

### PR DESCRIPTION
**Pull request by lxatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
- Create parameter
- Refactor Drag/Drop
- Refactor TemplateSelector
- Implement IDisposable on ThingDiagramContentItem

### CODE REVIEW

File: CDP4Composition/CommonView/Diagram/Cdp4DiagramOrgChartBehavior.cs [hunk 1]
```diff
@@ -518,14 +517,16 @@ private void AssociatedObjectOnItemsMoving(object sender, DiagramItemsMovingEven
         }
 
         /// <summary>
-        /// Delete related port shape when ever an element definition gets deleted
+        /// Delete  and related port shape when ever a <see cref="DiagramContentItem"/> gets deleted
```
Comment (samatrhea) [inline CDP4Composition/CommonView/Diagram/Cdp4DiagramOrgChartBehavior.cs:520]: spelling mistake? sentence doesn't make complete sense to me

**Review by samatrhea — CHANGES_REQUESTED** [review]

File: CDP4DiagramEditor/ViewModels/DiagramItemDataTemplateSelector.cs [hunk 2]
```diff
@@ -43,22 +60,13 @@ public class DiagramItemDataTemplateSelector : DataTemplateSelector
...
         public override DataTemplate SelectTemplate(object item, DependencyObject container)
         {
-            if (!(item is NamedThingDiagramContentItem))
+            return item switch
             {
-                return base.SelectTemplate(item, container);
-            }
-            
-            return item is DiagramPortViewModel ? this.DiagramPortTemplate : this.GenericDiagramItemDataTemplate;
+                ElementDefinitionDiagramContentItem => this.ElementBaseDiagramItemTemplate,
```
Comment (SimonWoodRhea) [inline CDP4DiagramEditor/ViewModels/DiagramItemDataTemplateSelector.cs:65]: In this situation its more usual to create a template in xaml per type using the template DataType parameter. Not sure if that's possible here

**Review by SimonWoodRhea — APPROVED** [review]

**Review by alexatrhea — APPROVED** [review]
nice!

File: CDP4DiagramEditor/ViewModels/DiagramItemDataTemplateSelector.cs (hunk 2 above)
Comment (lxatrhea) [reply CDP4DiagramEditor/ViewModels/DiagramItemDataTemplateSelector.cs:65]: DevExpress states that using a TemplateSelector is the way to go.

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**